src/plugins/data_source_management/public/components/create_data_source_wizard/components/create_form/create_data_source_form.tsx

```diff
@@ -109,8 +115,10 @@ export class CreateDataSourceForm extends React.Component<
     this.setState({ endpoint: e.target.value }, this.checkValidation);
   };
 
-  onChangeAuthType = (value: AuthType) => {
-    this.setState({ auth: { ...this.state.auth, type: value } }, this.checkValidation);
+  onChangeAuthType = (value: string) => {
+    const valueToSave =
+      value === AuthType.UsernamePasswordType ? AuthType.UsernamePasswordType : AuthType.NoAuth;
+    this.setState({ auth: { ...this.state.auth, type: valueToSave } }, this.checkValidation);
```

Comment:
I still prefer previous code with `value: AuthType`, why updating to use string?

Reply:
EuiRadio component onChange method will only give string as output. If we change type from string to AuthType, component will have error.